observability/opentelemetry/v2/http/http.go

```diff
@@ -0,0 +1,31 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package http
+
+import (
+	"net/http"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+
+	cehttp "github.com/cloudevents/sdk-go/v2/protocol/http"
+)
+
+// NewObservedHTTP creates an HTTP protocol with OTel trace propagating middleware.
+func NewObservedHTTP(opts ...cehttp.Option) (*cehttp.Protocol, error) {
+	// appends the OpenTelemetry Http transport + Middleware wrapper
+	// to properly trace outgoing and incoming requests from the client using this protocol
+	return cehttp.New(append(
+		[]cehttp.Option{
+			cehttp.WithRoundTripper(otelhttp.NewTransport(http.DefaultTransport)),
+			cehttp.WithMiddleware(func(next http.Handler) http.Handler {
+				return otelhttp.NewHandler(next, "", otelhttp.WithSpanNameFormatter(func(_ string, r *http.Request) string {
+					return "cloudevents.http.receiver." + r.URL.Path
```

Comment:
Yes I also thought about it. They have this as well in their OpenCensus impl: https://github.com/cloudevents/sdk-go/blob/main/observability/opencensus/v2/http/http.go#L25. What you think would be a good value? The outgoing events fro the client have a span name by default of `cloudevents.client.<eventtype> <op name>`, e.g. `cloudevents.client.my.event send` which should offer a good low cardinality. 

Maybe this could also follow the same? `cloudevents.http.<eventtype>` e.g. `cloudevents.http.my.event`?

Reply:
Went with `cloudevents.http.receiver`.